scripts/tracker.lua

```diff
@@ -55,11 +74,82 @@ function Tracker.newTracker(trackerType, trackerList)
         newTracker.userCanEnable = false
         newTracker.enabled = false
         newTracker.realtimeCamera = true
+    elseif trackerType == "cityblock" then
+        -- cityblock-specific data
+        newTracker.cityBlock = Tracker.cityBlock:new()
+        newTracker.minPos = { x = -5, y = -5 }
+        newTracker.maxPos = { x = 5, y = 5 }
+        Tracker.recalculateCityBlock(newTracker)
     end
 
     return newTracker
 end
 
+---comment  find the city block containing the position and relocate to that one
+---@param tracker Tracker.tracker
+---@param pos MapPosition.0|MapPosition.1
+function Tracker.focusCityBlock(tracker, pos)
+    local cityBlock = tracker.cityBlock
+    if cityBlock == nil then
+        return
+    end
+
+    cityBlock.currentBlock = {
+        x = math.floor((pos.x - cityBlock.blockOffset.x) / cityBlock.blockSize.x),
+        y = math.floor((pos.y - cityBlock.blockOffset.y) / cityBlock.blockSize.y),
+    }
+
+    Tracker.recalculateCityBlock(tracker)
+end
+
+---comment recalculates the tracker vital stats from the city block vital stats
+---@param tracker Tracker.tracker
+function Tracker.recalculateCityBlock(tracker)
```

Comment:
I would like to see some unit-tests on functions that perform calculations:
* it shows the meth is correct
* it makes changes/refactoring more easy

Reply:
I think this is the only outstanding issue. Do I add in a tracker-cityBlock.lua file and wire it into test-suite.lua?

I guess I'll have to think of a couple of tests that don't simply re-implement the city block logic :/

Reply:
That makes sense yes (having a tracker-cityblock.lua test file)

Calling the calculation function(s) and test with predefined/expected results is fine. Indeed, recalculating using the same method doesn't test much...

> I think this is the only outstanding issue

There are some small ones left I think (unless you fixed them but didn't push yet):
* https://github.com/veger/TLBE/pull/52#discussion_r1248968176
* https://github.com/veger/TLBE/pull/52#discussion_r1248963651